app/src/test/java/com/example/habitsmasher/ui/dashboard/DashboardFragmentTest.java

```diff
@@ -0,0 +1,32 @@
+package com.example.habitsmasher.ui.dashboard;
+
+import static org.junit.Assert.*;
+
+import com.example.habitsmasher.Habit;
+import com.example.habitsmasher.HabitList;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import org.junit.Test;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class DashboardFragmentTest{
+    DashboardFragment _dashboardFragment = new DashboardFragment();
+    SimpleDateFormat _formatter = new SimpleDateFormat("dd-MM-yyyy");
+    String _dateToString = "16-10-2021";
+    Date _date;
+
+    {
+        try {
+            _date = _formatter.parse(_dateToString);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+    @Test
+    public void addHabitSuccess(){
+        _dashboardFragment.addNewHabit("Habit title 1", "Habit reason 1", _date);
+    }
```

Comment:
We need to **assert** that the habit was actually added. This test isn't doing anything. It calls the method, but the whole point of a test is to **verify** the results after the operation you're testing.

The fields such as `_dateToString` and `Date`, along with the try/catch block can all be put locally within the test itself, as we may need to add other tests in the future which may use different dates/patterns.

Here is an example "add something to a list" unit test from the lab:
![image](https://user-images.githubusercontent.com/43623225/137645229-315b660b-3eb8-41af-9f60-21cf04276c34.png)